src/Game.tsx

```diff
@@ -1,93 +1,96 @@
-import * as React from "react";
+import React, { useState } from "react";
 import { TextField, Button, Fab } from "@material-ui/core";
 import SendIcon from "@material-ui/icons/Send";
 import RefreshIcon from "@material-ui/icons/Refresh";
-import { Dealer } from "./Dealer";
+import Deck from "card-deck";
+import { Team, PlayerInfo, Revealer } from "./Revealer";
 import "./Game.css";
 
-enum GameState {
-  OPTIONS = "options",
-  DEAL = "deal"
-}
+const NUM_SPIES_BY_PLAYER_COUNT: Record<number, number> = {
+  5: 2,
+  6: 2,
+  7: 3,
+  8: 3,
+  9: 3,
+  10: 4
+};
 
-interface State {
-  gameState: GameState;
-  selectedNumPlayers: number;
-}
+const dealPlayers = (numPlayers: number): PlayerInfo[] => {
+  const numSpies = NUM_SPIES_BY_PLAYER_COUNT[numPlayers];
+  const roles = new Array(numPlayers);
+  roles.fill(Team.SPY, 0, numSpies);
+  roles.fill(Team.RESISTANCE, numSpies);
 
-export class Game extends React.PureComponent<{}, State> {
-  private static NUM_PLAYERS_MIN = 5;
-  private static NUM_PLAYERS_MAX = 10;
+  const deck = new Deck(roles);
+  deck.shuffle();
 
-  state: State = {
-    gameState: GameState.OPTIONS,
-    selectedNumPlayers: Game.NUM_PLAYERS_MIN
-  };
+  const players = deck.draw(numPlayers).map((team: Team, i: number) => {
+    return {
+      name: `Player ${i + 1}`,
+      team
+    };
+  });
+  return players;
+};
 
-  componentDidMount(): void {
-    this.reset();
-  }
+export const Game: React.FC = () => {
+  const NUM_PLAYERS_MIN = 5;
+  const NUM_PLAYERS_MAX = 10;
 
-  private reset = (): void => this.setState({ gameState: GameState.OPTIONS });
+  const [selectedNumPlayers, setSelectedNumPlayers] = useState<number>(
+    NUM_PLAYERS_MIN
+  );
+  const [players, setPlayers] = useState<PlayerInfo[]>([]);
```

Comment:
I'd probably represent this as `PlayerInfo[] | undefined`, so that `undefined` can represent the uninitialized state. But that's primarily a stylistic thing.